spec/java/security/SecureRandom.main.lsl

```diff
@@ -0,0 +1,885 @@
+
+        val propertyLength: int = action CALL_METHOD(property, "length", []);
+        if (propertyLength == 0)
+            _throwNSAE();
+
+        val algorithms: array<String> = action CALL_METHOD(property, "split", [","]);
+        val algorithmsLength: int = action ARRAY_SIZE(algorithms);
+
+        var j: int = 0;
+        action LOOP_FOR(
+            j, 0, algorithmsLength, +1,
+            findAlg_loop(j, algorithms, result)
+        );
+
+        if (result == null)
+            _throwNSAE();
+    }
+
+
+    @Phantom proc findAlg_loop (j: int, algorithms: array<String>, result: SecureRandom): void
+    {
+        val curEntry: String = algorithms[j];
+        val algAndProv: array<String> = action CALL_METHOD(curEntry, "split", [":"]);
+        val algAndProvLength: int = action ARRAY_SIZE(algAndProv);
+        val _algorithm: String = algAndProv[0];
+
+        var resultProvider: Provider = null;
+        var resultAlgorithm: String = null;
+        val providersList: array<Provider> = action CALL_METHOD(null as Security, "getProviders", []);
+        val providersListLength: int = action ARRAY_SIZE(providersList);
+        var i: int = 0;
+
+        if (algAndProvLength == 2)
+        {
+            val providerName: String = algAndProv[1];
+
+            action LOOP_FOR(
+                i, 0, providersListLength, +1,
+                findAlgorithmByAllProvidersByName_loop(i, providersList, _algorithm, resultProvider, resultAlgorithm, providerName)
+            );
+        }
+        else
+        {
+            action LOOP_FOR(
+                i, 0, providersListLength, +1,
+                findAlgorithmByAllProviders_loop(i, providersList, _algorithm, resultProvider, resultAlgorithm)
+            );
+        }
+
+        if (resultAlgorithm != null)
+        {
+            val isDefaultProvider: boolean = _isDefaultProvider(resultProvider);
+
+            result = new SecureRandomAutomaton(state = Initialized,
+                provider = resultProvider,
+                algorithm = resultAlgorithm,
+                defaultProvider = isDefaultProvider,
+            );
+
+            action LOOP_BREAK();
+        }
+    }
+
+
+    @static fun *.getSeed (numBytes: int): array<byte>
+    {
+        if (numBytes < 0)
+            _throwIAE();
+
+        _generateSeed(result, numBytes);
+    }
+
+
+    // methods
+
+    // within java.util.Random
+    fun *.doubles (@target self: SecureRandom): DoubleStream
+    {
+        val mass: array<double> = _generateRandomDoubleArray(MAX_RANDOM_STREAM_SIZE);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new DoubleStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = MAX_RANDOM_STREAM_SIZE,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.doubles (@target self: SecureRandom, randomNumberOrigin: double, randomNumberBound: double): DoubleStream
```

Comment:
What if `randomNumberBound` is **less** than `randomNumberOrigin`?

Reply:
Fixed;